src/simplifiers/LA.cc

```diff
@@ -57,14 +57,14 @@ void LAExpression::initialize(PTRef e, bool do_canonize) {
         } else if (logic.isTimes(t)) {
             // If it is times, then one side must be constant, other
             // is enqueued with a new constant
-            auto [var, constant] = logic.splitTermToVarAndConst(t);
+            auto [var, constant] = logic.splitPolyTerm(t);
             Real new_c = logic.getNumConst(constant);
             new_c *= c;
             curr_term.emplace_back(var);
             curr_const.emplace_back(std::move(new_c));
         } else {
             // Otherwise it is a variable, Ite, UF or constant
-            assert(logic.isNumVarLike(t) || logic.isConstant(t) || logic.isUF(t));
+            assert(logic.isMonomial(t) || logic.isConstant(t) || logic.isUF(t));
```

Comment:
`isUF` should no longer be needed, no? It should be covered by the monomial check.
Also, please update the comment. For example, ITE is no longer possible here.

Reply:
Fixed